recipe/meta.yaml

```diff
@@ -1,64 +1,125 @@
-
-{% set vmajor,vminor,vpatch = version.split('.') %}
-# ALWAYS DOUBLE-CHECK THESE OFFSETS AGAINST UPSTREAM setup.py
-{% set botocore_major = vmajor | int + 0 %}
-{% set botocore_minor = vminor | int + 2 %}
-{% set botocore_patch = vpatch | int + 0 %}
-{% set botocore_pin = botocore_major ~ '.' ~ botocore_minor ~ '.' ~ botocore_patch %}
+{% set version = "2.8.11" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
+# v2 is not on PyPI https://github.com/aws/aws-cli/issues/6785
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.org/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ hash }}
+  url: https://github.com/aws/aws-cli/archive/{{ version }}.tar.gz
+  sha256: b37c64c6583153a420c7b80b34d62d33e317b40db561dfaeef027da083debb17
 
 build:
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv  # [not unix]
-  script: {{ PYTHON }} -m pip install . --no-deps -vv && mkdir -p $PREFIX/share/bash-completions/completions && echo complete -C aws_completer aws >$PREFIX/share/bash-completions/completions/aws  # [unix]
+  noarch: python
+  # Build uses pep517 with a custom backend
+  script: |
+    {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv
 
 requirements:
   build:
-    - python                                 # [build_platform != target_platform]
+    - python >=3.8                           # [build_platform != target_platform]
     - cross-python_{{ target_platform }}     # [build_platform != target_platform]
   host:
-    - python
-#    - python >=3.7
+    - python >=3.8
+    - flit-core
     - pip
 
+    # The build process generates the autocomplete index (scripts/gen-ac-index)
+    # which requires importing awscli modules, so we need to install all runtime
+    # dependencies too. YAML anchors don't support merging lists so we have to
+    # duplicate everything :-(
+    # Commented lines are the original requirements from
+    # https://github.com/aws/aws-cli/blob/2.8.11/pyproject.toml#L30-L40
+    # - colorama >=0.2.5,<0.4.4
+    - colorama >=0.2.5,<0.5.0
+    # - docutils >=0.10,<0.16
+    - docutils >=0.10,<0.18
+    - cryptography >=3.3.2,<=38.0.1
+    - ruamel.yaml >=0.15.0,<=0.17.21
+    - wcwidth <0.2.0
+    - prompt_toolkit >=3.0.24,<3.0.29
+    - distro >=1.5.0,<1.6.0
+    - awscrt >=0.12.4,<=0.14.0
+    - python-dateutil >=2.1,<3.0.0
+    - jmespath >=0.7.1,<1.1.0
+    - urllib3 >=1.25.4,<1.27
+
   run:
-    - python
-#    - python >=3.7
-    - botocore =={{ botocore_pin }}
-    - colorama >=0.2.5,<0.4.5
-    - docutils >=0.10,<0.17
-    - rsa >=3.1.2,<4.8
-    - s3transfer >=0.6.0,<0.7.0
-    - pyyaml >=3.10,<5.5
+    - python >=3.8
+    # Commented lines are the original requirements from
+    # https://github.com/aws/aws-cli/blob/2.8.11/pyproject.toml#L30-L40
+    # They have been expanded to work with the available conda-forge packages on
+    # more recent versions of Python
```

Comment:
please don't get ahead of upstream on any requirements

Reply:
OK, I've removed it.

Reply:
the good news is v2 turned on dependabot so they'll get automatic pr's updating these ranges (e.g. https://github.com/aws/aws-cli/pull/7402), I wonder whether there's enough infrastructure in the conda-forge autotick bot to detect those and update the yaml when upstream changes on a new release